nativelib/src/main/scala-next/scala/scalanative/safe/SafeZoneCompat.scala

```diff
@@ -0,0 +1,7 @@
+package scala.scalanative.safe
+
+import language.experimental.captureChecking
+
+object SafeZoneCompat {
+  def withSafeZone[T](sz: {*} SafeZone, obj: T): {sz} T = obj
```

Comment:
Instead of moving `withSafeZone` to `SafeZone`, maybe move it to `scala.scalanative.runtime` to indicate that it's an inner function and only the compiler should use it?

Reply:
Sure, we can do that as well, good point. It would be easiest to just move it there. At that point, we can also rename to `scala.scalanative.runtime.SafeZone`. 

Alternatively, we can make it a part of the `runtime` package object by usage of trait mixins and different sources in scala (empty trait) and scala-next (actual implementation available only with cc), but I don't think it's worth it.